atc/db/migration/migrations/1609959123_add_created_by_to_builds.up.sql

```diff
@@ -0,0 +1,4 @@
+BEGIN;
+ALTER TABLE builds
+    ADD COLUMN created_by text not null default ''::text;
```

Comment:
Should we just let this be `NULL`? I think setting a default will make the migration slow for deployments with a large number of builds, and I wouldn't expect any value to be present for builds which were created by Concourse.

Reply:
@vito  Actually I thought over that when I changed the field from `bson` to `text`. IMO, migration is an one-time effort, even if it's a little bit slow, it's acceptable. Even if a `builds` table contains millions of tuples, migration should be done within a few minutes (assuming that a large deployment must have a powerful db behind). In our case, we usually claim a maintenance window for each upgrade,  though Concourse may not really down. so I don't feel migration is a big problem.

On the other side, making this field `not null`, we will not need to scan this field into a `NullableString` first, which helps keep code clean, and is faster at runtime.

Reply:
I'm not super concerned about runtime performance of using `sql.NullString`; we already use it everywhere, and avoiding it for performance reasons seems like a micro-optimization. As for clean code, I don't think it's really that much worse; if you like we can keep things `string` everywhere instead of `*string` and handle the `NULL` value only in the one place where we read the value from the database.

I'm more concerned with having an precise representation to make troubleshooting easier; `""` vs. `NULL` can sometimes be a useful distinction. A `NULL` value means there was clearly no creator, while `""` could mean the operator-configured field was blank for the user in question (maybe they didn't configure their real name or email address).

Reply:
OK, I have made `created_by` nullable. And make API and `fly builds` to distinct "not set" and "empty:

* In API, if created by is not set, the field `created_by` absents. If the field is set with `""`, then it just takes `"created_by": ""`.
* In `fly builds`, if created by is not set, it shows "system".

<img width="1119" alt="fly-builds-w-system" src="https://user-images.githubusercontent.com/18585861/106706552-d13c9480-662a-11eb-8579-8b6c4562f203.png">

Reply:
Neat! That feels pretty clear. Thanks!